# Metis Final Project Part I. Gathering Article Links from the NYT API from 2000 to 2018.

### The Below code uses the requests library to access NYT Meta-data for the querieble search "United States Economy". After testing different searches out manually (i.e. searching the API for 'Business'), I determined this query to be sufficient.

In [2]:
import requests

In [237]:
queries = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&begin_date=20180801&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
articles = requests.get(queries).json()
for i in articles['response']['docs']: #print headlines (check relevancy)
    print(i['headline']['main'])

Was the U.S. Actually ‘Built on Tariffs,’ as Trump Says?
Worried About Turkey’s Economic Problems? China’s Could Be Worse
Stephen Miller’s Uncle Calls Him a Hypocrite in an Online Essay
You Know Who the Tax Cuts Helped? Rich People
The Trouble With Trucking
The Economy Grew Even Faster in Truman’s Presidency. So What?
Vanguard Warns of Worsening Odds for the Economy and Markets
An Economic Upturn Begun Under Obama Is Now Trump’s to Tout
Looking Back at the Economic Crash of 2008
The Toll of America’s Obesity 


In [285]:
queries = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20180801&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
articles = requests.get(queries).json()
for i in articles['response']['docs']: #print headlines (check relevancy)
    print(i['headline']['main'])
    print(i['pub_date'])

How Big a Deal Is Trump’s Fuel Economy Rollback? For the Climate, Maybe the Biggest Yet
2018-08-03T21:14:27+0000
Why a Trade Truce Could Add Almost $2 Trillion to the Stock Market
2018-08-03T19:32:16+0000
Buoyant Economy or a Blip? 4 Tips for Investing Before the Party Ends
2018-08-03T17:26:48+0000
Trump’s Economic Claims Are Overblown
2018-08-03T15:04:11+0000
Workers Hardest Hit by Recession Are Joining in Recovery
2018-08-03T09:00:05+0000
Apple’s $1 Trillion Milestone Reflects Rise of Powerful Megacompanies
2018-08-02T23:29:10+0000
What Will Cause the Next Recession? A Look at the 3 Most Likely Possibilities
2018-08-02T09:18:28+0000
Congress Strengthens Reviews of Chinese and Other Foreign Investments
2018-08-02T00:55:29+0000
Fed Holds Interest Rates Steady but, Defying Trump, Signals Increases to Come
2018-08-01T09:00:05+0000
Losing Earth: The Decade We Almost Stopped Climate Change
2018-08-01T04:47:33+0000


In [530]:
from random import randint
from time import sleep

url = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [531]:
nyt_list_1 = nyt_get(url)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=6&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=51&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=52&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=53&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=54&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=55&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=56&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
20

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=101&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=102&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=103&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=104&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=105&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=106&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=150&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=151&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=152&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=153&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=154&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=155&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=199&begin_date=20100103&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200


In [532]:
nyt_list_1[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '55cd96e438f0d85373a17a4f',
    'blog': {},
    'byline': {'organization': None,
     'original': 'By STEVEN RATTNER',
     'person': [{'firstname': 'Steven',
       'lastname': 'RATTNER',
       'middlename': None,
       'organization': '',
       'qualifier': None,
       'rank': 1,
       'role': 'reported',
       'title': None}]},
    'document_type': 'article',
    'headline': {'content_kicker': 'Contributing Op-Ed Writer',
     'kicker': 'Contributing Op-Ed Writer',
     'main': 'Trump’s Economic Muddle',
     'name': None,
     'print_headline': 'Trump’s Economic Muddle  ',
     'seo': None,
     'sub': None},
    'keywords': [{'major': 'N',
      'name': 'persons',
      'rank': 1,
      'value': 'Trump, Donald J'},
     {'major': 'N',
      'name': 'subject',
      'rank': 2,
      'value': 'United States Economy'},
     {'major': 'N',
      'name': 'subject',


### The above code accesses article meta-data content between pages 1 to 199 feautring the search 'United States Economy

### Looking at the last page from the list used in the function above, the last publishing article is in August-9th  of 2015.  The day before will now be used as the end for the next API-URL grab, choosing an arbitrary date further in the past as a begin date. 

In [305]:
from random import randint
from time import sleep

url1 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get1(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [306]:
nyt_list_2 = nyt_get1(url1)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=135&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=179&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=180&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=20120101&end_date=201508007&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United

In [315]:
nyt_list_2[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '526b08d338f0d8384ec1f055',
    'abstract': "Commerce Dept reports orders for durable goods rose 3.7 percent in September, above the 0.2 percent August gain; University of Michigan reports its index of consumer sentiment fell to 73.2 from 77.5 in September as concern grew over the shutdown and political fight over nation's borrowing limit.",
    'blog': {},
    'byline': {'organization': 'THE ASSOCIATED PRESS',
     'original': 'By THE ASSOCIATED PRESS',
     'person': []},
    'document_type': 'article',
    'headline': {'content_kicker': None,
     'kicker': None,
     'main': 'Orders for Durable Goods Rise, but Report Shows Sliding Consumer Sentiment',
     'name': None,
     'print_headline': 'Orders for Durable Goods Rise, but Report Shows Sliding Consumer Sentiment',
     'seo': None,
     'sub': None},
    'keywords': [{'major': None,
      'name': 'subject',
     

### The last date generated for the previous search was 2013-10-23

In [316]:
from random import randint
from time import sleep

url2 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get2(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [317]:
nyt_list_3 = nyt_get2(url2)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=20090101&end_date=20131023&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

In [321]:
nyt_list_3[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '53f91cfa38f0d84410bcab89',
    'abstract': 'Grim news, ranging from weak corporate earnings to a deepening European debt crisis, leads investors to a sell-off; Dow Jones industrial average experiences its third straight triple-digit loss; worldwide investors continue to sell stocks and seek the relative safety of United States government debt, sending the yield on the benchmark 10-year Treasury note to another record low.',
    'blog': {},
    'byline': {'organization': 'THE ASSOCIATED PRESS',
     'original': 'By THE ASSOCIATED PRESS',
     'person': []},
    'document_type': 'article',
    'headline': {'content_kicker': 'Stocks and Bonds',
     'kicker': None,
     'main': 'With Little Good News, Wall Street Pulls Back',
     'name': None,
     'print_headline': 'With Little Good News, Wall Street Pulls Back',
     'seo': None,
     'sub': None},
    'keywords': [{'maj

### Date for the last pull was 2012-07-21

In [322]:
from random import randint
from time import sleep

url3 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get3(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [323]:
nyt_list_4 = nyt_get3(url3)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=20080101&end_date=20120720&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

In [327]:
nyt_list_4[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '5410fbec38f0d80f29521fe8',
    'blog': {},
    'byline': {'organization': None,
     'original': 'By ERIC DASH',
     'person': [{'firstname': 'Eric',
       'lastname': 'DASH',
       'middlename': None,
       'organization': '',
       'qualifier': None,
       'rank': 1,
       'role': 'reported',
       'title': None}]},
    'document_type': 'article',
    'headline': {'content_kicker': None,
     'kicker': None,
     'main': 'AAA Rating Is a Rarity in Business',
     'name': None,
     'print_headline': 'AAA Rating Is a Rarity In Business',
     'seo': None,
     'sub': None},
    'keywords': [{'major': None,
      'name': 'subject',
      'rank': 1,
      'value': 'Ratings and Rating Systems'},
     {'major': None,
      'name': 'subject',
      'rank': 2,
      'value': 'United States Economy'},
     {'major': None, 'name': 'subject', 'rank': 3, 'value': 'Credit 

### Date for last pull for above was 2011/08/02

In [330]:
from random import randint
from time import sleep

url4 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get4(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [331]:
nyt_list_5 = nyt_get4(url4)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=20070101&end_date=20110801&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

In [333]:
nyt_list_5[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '5445f40338f0d832c813f794',
    'blog': {},
    'byline': {'organization': None,
     'original': 'By MOTOKO RICH',
     'person': [{'firstname': 'Motoko',
       'lastname': 'RICH',
       'middlename': None,
       'organization': '',
       'qualifier': None,
       'rank': 1,
       'role': 'reported',
       'title': None}]},
    'document_type': 'article',
    'headline': {'content_kicker': '',
     'kicker': '',
     'main': 'Economy Gains\r\nImpetus as U.S.\r\nAdds 290,000 Jobs',
     'name': None,
     'print_headline': 'U.S. Shows Surprising Gain in Jobs',
     'seo': None,
     'sub': None},
    'keywords': [{'major': 'N',
      'name': 'subject',
      'rank': 1,
      'value': 'United States Economy'},
     {'major': 'N', 'name': 'subject', 'rank': 2, 'value': 'Labor and Jobs'},
     {'major': 'N', 'name': 'subject', 'rank': 3, 'value': 'Unemployment'}],
    

### Last date for pull above was 2010/05/06

In [336]:
from random import randint
from time import sleep

url5 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get5(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [337]:
nyt_list_6 = nyt_get5(url5)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=20060101&end_date=20100505&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

In [339]:
nyt_list_6[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '545db5ab38f0d877bc740c91',
    'blog': {},
    'byline': {'organization': None,
     'original': 'By DAVID BROOKS',
     'person': [{'firstname': 'David',
       'lastname': 'BROOKS',
       'middlename': None,
       'organization': '',
       'qualifier': None,
       'rank': 1,
       'role': 'reported',
       'title': None}]},
    'document_type': 'article',
    'headline': {'content_kicker': 'Op-Ed Columnist',
     'kicker': 'Op-Ed Columnist',
     'main': 'The Worst-Case Scenario',
     'name': None,
     'print_headline': 'The Worst-Case Scenario',
     'seo': None,
     'sub': None},
    'keywords': [{'major': 'N',
      'name': 'subject',
      'rank': 1,
      'value': 'United States Economy'},
     {'major': 'N',
      'name': 'subject',
      'rank': 2,
      'value': 'Subprime Mortgage Crisis'},
     {'major': 'N', 'name': 'persons', 'rank': 3, 'value': 'Ob

### Last date for pull above was 2009/02/11

In [341]:
from random import randint
from time import sleep

url6 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get6(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [342]:
nyt_list_7 = nyt_get6(url6)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=20050101&end_date=20090210&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

In [344]:
nyt_list_7[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '548056a838f0d82816288aeb',
    'blog': {},
    'byline': {'organization': None,
     'original': 'By MICHAEL M. GRYNBAUM',
     'person': [{'firstname': 'Michael',
       'lastname': None,
       'middlename': None,
       'organization': '',
       'qualifier': None,
       'rank': 1,
       'role': 'reported',
       'title': None}]},
    'document_type': 'article',
    'headline': {'content_kicker': None,
     'kicker': None,
     'main': 'In Fed Report, Signs of Continued Weakness',
     'name': None,
     'print_headline': 'Signs of Continued Weakness in Fed Report',
     'seo': None,
     'sub': None},
    'keywords': [{'major': None,
      'name': 'subject',
      'rank': 1,
      'value': 'Consumer Price Index'},
     {'major': None,
      'name': 'subject',
      'rank': 2,
      'value': 'United States Economy'}],
    'multimedia': [],
    'news_desk': 'Busines

### Last date for pull above was 2008/04/15

In [345]:
from random import randint
from time import sleep

url7 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get7(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [346]:
nyt_list_8 = nyt_get7(url7)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=20040101&end_date=20080415&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

In [353]:
nyt_list_8[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '4fd25c838eb7c8105d80a47f',
    'blog': {},
    'byline': {'organization': None,
     'original': 'By ANNA BERNASEK',
     'person': [{'firstname': 'Anna',
       'lastname': 'BERNASEK',
       'middlename': None,
       'organization': '',
       'qualifier': None,
       'rank': 1,
       'role': 'reported',
       'title': None}]},
    'document_type': 'article',
    'headline': {'content_kicker': None,
     'kicker': 'ECONOMIC VIEW',
     'main': "A Number That's Meant to Be Second-Guessed",
     'name': None,
     'print_headline': None,
     'seo': None,
     'sub': None},
    'keywords': [{'major': None,
      'name': 'persons',
      'rank': 0,
      'value': 'GREENSPAN, ALAN'},
     {'major': None,
      'name': 'organizations',
      'rank': 0,
      'value': 'FEDERAL RESERVE SYSTEM'},
     {'major': None,
      'name': 'organizations',
      'rank': 0,
      'v

### Last date for pull above was 2005/07/28

In [354]:
from random import randint
from time import sleep

url8 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get8(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [355]:
nyt_list_9 = nyt_get8(url8)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=19980101&end_date=20050727&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

### Last date for pull above was 2002-09-22

In [359]:
print("TEST")

TEST


In [360]:
len(nyt_list_8)

199

In [361]:
len(nyt_list_9)

199

In [378]:
nyt_list_2[0]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '55bd12b038f0d8588a9d055b',
    'abstract': "Editorial takes issue with Federal Reserve's optimistic outlook on United States economy; cautions Fed that if it intends to increase interest rates it must make case that current economic conditions are as good as can be.",
    'blog': {},
    'byline': {'organization': 'THE EDITORIAL BOARD',
     'original': 'By THE EDITORIAL BOARD',
     'person': []},
    'document_type': 'article',
    'headline': {'content_kicker': 'Editorial',
     'kicker': 'Editorial',
     'main': 'Why the Economy Is Not Ready for an Interest Rate Increase',
     'name': None,
     'print_headline': 'The Ho-Hum Economy',
     'seo': None,
     'sub': None},
    'keywords': [{'major': None,
      'name': 'subject',
      'rank': 1,
      'value': 'Editorials'},
     {'major': None,
      'name': 'subject',
      'rank': 2,
      'value': 'United States

In [379]:
with open('nyt_list_2.pickle', 'wb') as f2:
    pickle.dump(nyt_list_2, f2)

In [389]:
with open('nyt_list_3.pickle', 'wb') as f3:
    pickle.dump(nyt_list_3, f3)

In [390]:
with open('nyt_list_4.pickle', 'wb') as f4:
    pickle.dump(nyt_list_4, f4)

In [391]:
with open('nyt_list_5.pickle', 'wb') as f5:
    pickle.dump(nyt_list_5, f5)

In [392]:
with open('nyt_list_6.pickle', 'wb') as f6:
    pickle.dump(nyt_list_6, f6)

In [393]:
with open('nyt_list_7.pickle', 'wb') as f7:
    pickle.dump(nyt_list_7, f7)

In [394]:
with open('nyt_list_8.pickle', 'wb') as f8:
    pickle.dump(nyt_list_8, f8)

In [395]:
with open('nyt_list_9.pickle', 'wb') as f9:
    pickle.dump(nyt_list_9, f9)

In [405]:
with open('nyt_list_10.pickle', 'wb') as f10:
    pickle.dump(nyt_list_10, f10)

In [399]:
with open('nyt_list_1.pickle', 'wb') as f1:
    pickle.dump(nyt_list_1, f1)

In [401]:
from random import randint
from time import sleep

url9 = "http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page={}&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352"
def nyt_get9(url):
    #begin_date=20100103 is the date of half the guardian articles
    nyt_list = []

        
    for i in range(1,200):

        #url = url.format(i)
        r = requests.get(url.format(i))
        sleep(randint(1,2))
        print(url.format(i))
        print(r.status_code)
        if r.status_code == 200:
            
            nyt_list.append(r.json())
            
        else:
            print('cant pull no more :(')

    return nyt_list

In [402]:
nyt_list_10 = nyt_get9(url9)

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=1&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=2&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=3&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=4&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=5&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=46&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=47&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=48&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=49&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=50&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=91&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=92&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=93&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=94&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=95&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Ec

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=136&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=137&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=138&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=139&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=140&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=181&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=182&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=183&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=184&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United States Economy'&page=185&begin_date=19950101&end_date=20020921&sort=newest&api-key=34ae6398b29445188f15963b07e30352
200
http://api.nytimes.com/svc/search/v2/articlesearch.json?q='United Stat

In [404]:
nyt_list_10[198]

{'copyright': 'Copyright (c) 2018 The New York Times Company. All Rights Reserved.',
 'response': {'docs': [{'_id': '59c9284c95d0e0246f20c195',
    'abstract': 'In this lesson, students explore how unions and business groups in the United States try to influence Congress and the American people in deciding whether or not to accord China permanent normal trading status. Students then work in groups to create commercials convincing viewers that China should or should not receive this special status.',
    'blog': {},
    'byline': {'organization': None,
     'original': 'By KARI KOHL',
     'person': [{'firstname': 'Kari',
       'lastname': 'KOHL',
       'middlename': None,
       'organization': '',
       'qualifier': None,
       'rank': 1,
       'role': 'reported',
       'title': None}]},
    'document_type': 'blogpost',
    'headline': {'content_kicker': None,
     'kicker': 'The Learning Network',
     'main': 'Trading Places',
     'name': None,
     'print_headline': None,
  

In [418]:
print(nyt_list_10[198]['response']['docs'][9]['web_url'])#['webUrl'])

https://www.nytimes.com/2000/05/18/business/the-markets-bonds-prices-of-treasuries-decline-as-investors-assess-fed-move.html


### Last date for pull above was 05/18/2000

## The above code extracts meta-data from the NYT API. 
## The below code extracts each individual link from the metadata. The links are then appended to a blank list. 

In [421]:
nyt_list_10_urls = []
for item in nyt_list_10:
    nyt_list_10_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [422]:
nyt_list_10_urls

['https://www.nytimes.com/2002/09/12/business/fed-reports-a-recent-slowing-of-the-economy.html',
 'https://www.nytimes.com/2002/09/12/business/the-markets-currencies.html',
 'https://www.nytimes.com/2002/09/10/opinion/the-long-haul.html',
 'https://www.nytimes.com/2002/09/10/business/inventories-rose-in-july-and-shoppers-added-to-debt.html',
 'https://www.nytimes.com/2002/09/09/us/long-term-jobless-rose-by-50-percent-over-the-last-year.html',
 'https://www.nytimes.com/2002/09/08/business/l-cutting-corporate-costs-or-playing-with-fire-769320.html',
 'https://www.nytimes.com/2002/09/08/business/portfolios-etc-numbers-look-better-but-markets-see-the-worst.html',
 'https://www.nytimes.com/2002/09/08/business/l-cutting-corporate-costs-or-playing-with-fire-769339.html',
 'https://www.nytimes.com/2002/09/08/business/c-corrections-768022.html',
 'https://www.nytimes.com/2002/09/08/business/l-cutting-corporate-costs-or-playing-with-fire-769312.html',
 'https://www.nytimes.com/2002/09/08/busines

In [424]:
len(set(nyt_list_10_urls))

1984

In [425]:
nyt_list_9_urls = []
for item in nyt_list_9:
    nyt_list_9_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [431]:
(nyt_list_9_urls)

['https://www.nytimes.com/2005/07/18/opinion/18greider.html',
 'https://www.nytimes.com/2005/07/17/business/yourmoney/17port.html',
 'https://www.nytimes.com/2005/07/15/opinion/15deutch.html',
 'https://www.nytimes.com/2005/07/15/business/15stox.html',
 'https://www.nytimes.com/2005/07/14/business/14income.html',
 'https://www.nytimes.com/2005/07/13/business/worldbusiness/13unocal.html',
 'https://www.nytimes.com/2005/07/12/business/worldbusiness/12oil.ready.html',
 'https://www.nytimes.com/2005/07/10/weekinreview/10gross.html',
 'https://www.nytimes.com/2005/07/10/business/yourmoney/10fore.html',
 'https://www.nytimes.com/2005/07/10/business/mutfund/10lede.html',
 'https://www.nytimes.com/2005/07/09/business/worldbusiness/09five.ready.html',
 'https://www.nytimes.com/2005/07/09/business/worldbusiness/09markets.html',
 'https://www.nytimes.com/2005/07/09/business/09econ.html',
 'https://www.nytimes.com/2005/07/09/business/09oil.html',
 'https://www.nytimes.com/2005/07/08/business/08sho

In [428]:
nyt_list_8_urls = []
for item in nyt_list_8:
    nyt_list_8_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [430]:
len(set(nyt_list_8_urls))

1985

In [432]:
nyt_list_7_urls = []
for item in nyt_list_7:
    nyt_list_7_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [433]:
nyt_list_7_urls

['https://www.nytimes.com/2009/02/10/washington/10stimulus-web.html',
 'https://www.nytimes.com/2009/02/09/opinion/09krugman.html',
 'https://www.nytimes.com/2009/02/09/business/09bailout.html',
 'https://learning.blogs.nytimes.com/2009/02/09/a-plan-for-recovery/',
 'https://www.nytimes.com/2009/02/09/business/09ahead.html',
 'https://www.nytimes.com/2009/02/09/us/politics/09obama.html',
 'https://www.nytimes.com/2009/02/09/us/politics/09econ.html',
 'https://www.nytimes.com/2009/02/09/business/09junket.html',
 'https://www.nytimes.com/2009/02/09/us/politics/09stimulus.html',
 'https://www.nytimes.com/2009/02/08/opinion/08rich.html',
 'https://www.nytimes.com/2009/02/08/opinion/08sun1.html',
 'https://www.nytimes.com/2009/02/08/business/yourtaxes/08gray.html',
 'https://www.nytimes.com/2009/02/08/business/yourtaxes/08entre.html',
 'https://www.nytimes.com/2009/02/08/business/yourtaxes/08essay.html',
 'https://www.nytimes.com/2009/02/08/business/yourtaxes/08tips.html',
 'https://www.nyt

In [435]:
len(set(nyt_list_7_urls))

1978

In [436]:
nyt_list_6_urls = []
for item in nyt_list_6:
    nyt_list_6_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [438]:
nyt_list_6_urls

['https://www.nytimes.com/2010/04/30/business/economy/30econ.html',
 'https://www.nytimes.com/2010/04/29/us/politics/29fiscal.html',
 'https://www.nytimes.com/2010/04/29/business/29fed.html',
 'https://query.nytimes.com/gst/fullpage.html?res=9404E7DF163BF93BA15757C0A9669D8B63',
 'https://www.nytimes.com/2010/04/28/opinion/lweb28cohan.html',
 'https://www.nytimes.com/2010/04/28/opinion/l28goldman.html',
 'https://www.nytimes.com/2010/04/28/us/politics/28obama.html',
 'https://www.nytimes.com/2010/04/28/business/economy/28econ.html',
 'https://query.nytimes.com/gst/fullpage.html?res=9C00EED81139F934A15757C0A9669D8B63',
 'https://dealbook.nytimes.com/2010/04/26/a-change-of-topics-for-pandit-world-growth/',
 'https://www.nytimes.com/2010/04/27/business/27home.html',
 'https://dealbook.nytimes.com/2010/04/26/a-change-of-topics-for-pandit-world-growth/',
 'https://www.nytimes.com/2010/04/26/business/26ahead.html',
 'https://www.nytimes.com/2010/04/26/business/economy/26econ.html',
 'https://

In [439]:
len(set(nyt_list_6_urls))

1952

In [440]:
nyt_list_5_urls = []
for item in nyt_list_5:
    nyt_list_5_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [442]:
len(set(nyt_list_5_urls))

1950

In [443]:
nyt_list_5_urls

['https://www.nytimes.com/2011/07/31/business/with-threat-of-us-credit-downgrade-visions-of-apocalypse-and-zombies.html',
 'https://www.nytimes.com/2011/07/31/business/economy/whats-with-all-the-bernanke-bashing.html',
 'https://www.nytimes.com/2011/07/31/business/economy/sure-cure-for-debt-problems-is-economic-growth.html',
 'https://www.nytimes.com/2011/07/31/business/economy/taking-a-closer-look-at-a-downgrades-effects.html',
 'https://www.nytimes.com/interactive/2011/07/30/opinion/sunday/20110731_McFadden_Cartoon.html',
 'https://www.nytimes.com/2011/07/30/us/politics/30obama.html',
 'https://www.nytimes.com/2011/07/30/business/global/global-concern-over-us-debt-ceiling-disagreement.html',
 'https://opinionator.blogs.nytimes.com/2011/07/29/scold-your-own/',
 'https://economix.blogs.nytimes.com/2011/07/29/podcast-g-d-p-european-stocks-bernanke-and-bing/',
 'https://economix.blogs.nytimes.com/2011/07/29/ranking-the-presidents-by-g-d-p/',
 'https://www.nytimes.com/2011/07/30/business/

In [444]:
nyt_list_4_urls = []
for item in nyt_list_4:
    nyt_list_4_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [445]:
len(set(nyt_list_4_urls))

1900

In [446]:
nyt_list_4_urls

['https://www.nytimes.com/2012/07/19/business/economy/us-housing-starts-near-a-four-year-high.html',
 'https://www.nytimes.com/2012/07/19/technology/ericsson-earnings-decline-63-percent-in-2d-quarter.html',
 'https://www.nytimes.com/2012/07/18/us/politics/romney-and-obama-resume-economic-attacks.html',
 'https://www.nytimes.com/2012/07/18/opinion/the-world-as-a-fishbowl.html',
 'https://www.nytimes.com/2012/07/18/opinion/modern-capitalism-as-a-campaign-issue.html',
 'https://thecaucus.blogs.nytimes.com/2012/07/17/in-texas-a-presidential-duel-over-the-economy/',
 'https://boss.blogs.nytimes.com/2012/07/17/drilling-down-jared-bernstein-says-small-businesses-are-not-overtaxed/',
 'https://www.nytimes.com/2012/07/18/us/in-report-on-states-finances-a-grim-long-term-forecast.html',
 'https://www.nytimes.com/2012/07/18/business/economy/bernanke-testifies-before-senate-panel.html',
 'https://www.nytimes.com/2012/07/18/business/daily-stock-market-activity.html',
 'https://www.nytimes.com/2012/0

In [447]:
nyt_list_3_urls = []
for item in nyt_list_3:
    nyt_list_3_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [448]:
len(set(nyt_list_3_urls))

1989

In [449]:
nyt_list_3_urls

['https://www.nytimes.com/2013/10/23/business/daily-stock-market-activity.html',
 'https://economix.blogs.nytimes.com/2013/10/22/the-employment-rate-remains-stuck/',
 'https://economix.blogs.nytimes.com/2013/10/21/in-belated-jobs-report-a-search-for-clues/',
 'https://dealbook.nytimes.com/2013/10/21/economic-theory-via-youtube-and-cartoon/',
 'https://economix.blogs.nytimes.com/2013/10/21/medicaid-and-the-incentive-to-work/',
 'https://www.nytimes.com/2013/10/22/business/economy/sales-of-existing-homes-slid-back-in-september.html',
 'https://economix.blogs.nytimes.com/2013/10/21/economic-data-starts-flowing-again/',
 'https://economix.blogs.nytimes.com/2013/10/21/alan-greenspan-sees-inflation/',
 'https://boss.blogs.nytimes.com/2013/10/21/today-in-small-business-drywall-contractors/',
 'https://economix.blogs.nytimes.com/2013/10/21/the-inefficient-market-hypothesis/',
 'https://opinionator.blogs.nytimes.com/2013/10/20/questions-for-free-market-moralists/',
 'https://www.nytimes.com/201

In [450]:
nyt_list_2_urls = []
for item in nyt_list_2:
    nyt_list_2_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [451]:
len(set(nyt_list_2_urls))

1990

In [452]:
nyt_list_2_urls

['https://www.nytimes.com/2015/08/02/opinion/sunday/why-the-economy-is-not-ready-for-an-interest-rate-increase.html',
 'https://www.nytimes.com/2015/08/02/education/edlife/liberal-arts-a-lost-cause.html',
 'https://www.nytimes.com/2015/08/02/opinion/sunday/were-making-life-too-hard-for-millennials.html',
 'https://www.nytimes.com/2015/07/31/business/economy/us-economy-q2-gdp-first-estimate.html',
 'https://www.nytimes.com/2015/07/31/business/international/pacific-trade-deal-drugs-patent-protection.html',
 'https://www.nytimes.com/2015/07/30/business/economy/limited-supplies-hold-back-june-home-sales.html',
 'https://www.nytimes.com/2015/07/30/business/economy/federal-reserve-meeting-interest-rates.html',
 'https://www.nytimes.com/2015/07/30/business/daily-stock-market-activity.html',
 'https://news.blogs.nytimes.com/2015/07/29/morning-agenda-a-flutter-over-twitters-growth/',
 'https://www.nytimes.com/2015/07/29/upshot/health-spending-forecast-no-drastic-rise-but-slowdown-seems-over.htm

In [533]:
nyt_list_1_urls = []
for item in nyt_list_1:
    nyt_list_1_urls.extend([item['response']['docs'][idx]['web_url'] for idx in range(10)])

In [534]:
len(set(nyt_list_1_urls))

1990

In [455]:
nyt_list_1_urls

['https://www.nytimes.com/2018/08/10/books/review/adam-tooze-crashed.html',
 'https://www.nytimes.com/2018/08/09/opinion/cost-diabetes-obesity-budget.html',
 'https://www.nytimes.com/2018/08/09/opinion/corporate-debt-bubble-next-recession.html',
 'https://www.nytimes.com/2018/08/08/opinion/trump-tariffs-china-trade-war-who-will-win.html',
 'https://www.nytimes.com/2018/08/08/books/review-crashed-financial-crisis-adam-tooze.html',
 'https://www.nytimes.com/2018/08/08/technology/personaltech/fred-economics-writers.html',
 'https://www.nytimes.com/2018/08/07/us/politics/legal-immigrants-welfare-republicans-trump.html',
 'https://www.nytimes.com/2018/08/06/opinion/clouds-darken-trumps-sunny-economic-view.html',
 'https://www.nytimes.com/2018/08/06/reader-center/american-unemployment.html',
 'https://www.nytimes.com/2018/08/05/opinion/columnists/trump-real-wages-inflation-workers-economy.html',
 'https://www.nytimes.com/2018/08/05/opinion/notes-on-a-butter-republic.html',
 'https://www.nyti

In [512]:
nyt_list_1_urls

['https://www.nytimes.com/2018/08/10/books/review/adam-tooze-crashed.html',
 'https://www.nytimes.com/2018/08/09/opinion/cost-diabetes-obesity-budget.html',
 'https://www.nytimes.com/2018/08/09/opinion/corporate-debt-bubble-next-recession.html',
 'https://www.nytimes.com/2018/08/08/opinion/trump-tariffs-china-trade-war-who-will-win.html',
 'https://www.nytimes.com/2018/08/08/books/review-crashed-financial-crisis-adam-tooze.html',
 'https://www.nytimes.com/2018/08/08/technology/personaltech/fred-economics-writers.html',
 'https://www.nytimes.com/2018/08/07/us/politics/legal-immigrants-welfare-republicans-trump.html',
 'https://www.nytimes.com/2018/08/06/opinion/clouds-darken-trumps-sunny-economic-view.html',
 'https://www.nytimes.com/2018/08/06/reader-center/american-unemployment.html',
 'https://www.nytimes.com/2018/08/05/opinion/columnists/trump-real-wages-inflation-workers-economy.html',
 'https://www.nytimes.com/2018/08/05/opinion/notes-on-a-butter-republic.html',
 'https://www.nyti

In [535]:

nyt_list_oneten_urls = nyt_list_1_urls+nyt_list_2_urls+nyt_list_3_urls+\
                                       nyt_list_4_urls+nyt_list_5_urls+nyt_list_6_urls+\
                                       nyt_list_7_urls+nyt_list_8_urls+nyt_list_9_urls+\
                                       nyt_list_10_urls

### List nyt_list_oneten_urls contains NYT article url's from 2000-2018

In [537]:
len(nyt_list_oneten_urls)

19900

In [538]:
len(set(nyt_list_oneten_urls))

19702

In [539]:
completelist = set(nyt_list_oneten_urls)

In [540]:
len(completelist)

19702

In [542]:
nyt_list_oneten_urls

['https://www.nytimes.com/2018/08/10/books/review/adam-tooze-crashed.html',
 'https://www.nytimes.com/2018/08/09/opinion/cost-diabetes-obesity-budget.html',
 'https://www.nytimes.com/2018/08/09/opinion/corporate-debt-bubble-next-recession.html',
 'https://www.nytimes.com/2018/08/08/opinion/trump-tariffs-china-trade-war-who-will-win.html',
 'https://www.nytimes.com/2018/08/08/books/review-crashed-financial-crisis-adam-tooze.html',
 'https://www.nytimes.com/2018/08/08/technology/personaltech/fred-economics-writers.html',
 'https://www.nytimes.com/2018/08/07/us/politics/legal-immigrants-welfare-republicans-trump.html',
 'https://www.nytimes.com/2018/08/06/opinion/clouds-darken-trumps-sunny-economic-view.html',
 'https://www.nytimes.com/2018/08/06/reader-center/american-unemployment.html',
 'https://www.nytimes.com/2018/08/05/opinion/columnists/trump-real-wages-inflation-workers-economy.html',
 'https://www.nytimes.com/2018/08/05/opinion/notes-on-a-butter-republic.html',
 'https://www.nyti

In [544]:
with open('completelist.pickle', 'wb') as ff:
    pickle.dump(completelist, ff)

In [543]:
len(completelist)

19702

### Once the set function is performed above, the unique list of URL's is stored as a pickle file titled complete list. The list of URLs contains 19,702 unique items. 